feat(doc): publish documentation guidelines (#3576)
scaleway/docs-content: guidelines.mdx

---
meta:
  title: Scaleway Documentation Guidelines
  description: Discover comprehensive guidelines on writing and formatting for our documentation site. Ensure consistency and clarity with our style guide, covering grammar, terminology, and best practices.
content:
  h1: Scaleway Documentation Guidelines
  paragraph: Discover comprehensive guidelines on writing and formatting for our documentation site. Ensure consistency and clarity with our style guide, covering grammar, terminology, and best practices.

> tags: style-guide guidelines wording grammar

Author: Should we add validation/posted date?

Reviewer: Validation date might be useful to showcase the document itself isn't outdated.
Posted may not be necessary, I think.

| ✅ AMERICAN ENGLISH   | ❌ OTHER VARIATIONS   |
| --------------------- | --------------------- |
| Center, fiber         | Centre, fibre         |
| Recognize, prioritize | Recognise, prioritise |
| Favorite, color       | Favourite, colour     |

We do not accept contributions to Scaleway Documentation in languages other than English.

### Tone

The tone of Scaleway Documentation is generally **formal**. Do not use the **passive voice**, **"please"** or **"thank you"**, **the future tense** or **contractions**.

| ✅ DO WRITE                                | ❌ DON'T WRITE                                  |
| ------------------------------------------ | ----------------------------------------------- |
| Click **DELETE**  to delete the Instance.  | The Instance is deleted by clicking **DELETE**. |
| Tick the box to accept the conditions.     | Please tick the box to accept the conditions.   |
| Select **Metrics**. The Grafana dashboard opens. | Select **Metrics**. The Grafana dashboard will open. |

> | You cannot leave the field empty.          | You can't leave the field empty.                |

Reviewer: With this example, shouldn't be "DON'T WRITE" -> "DO NOT WRITE" ?

- For Scaleway design and branding guidelines, see [Ultraviolet](https://ultraviolet.scaleway.com/6dd9b5c45/p/62b4e2-ultraviolet).

## Overall writing guidelines

### Language

Scaleway Documentation uses [American English](https://en.wikipedia.org/wiki/American_English) (US English). Where words are spelled differently in American English to, for example, British English, use the American spelling. Here are some examples:

| ✅ AMERICAN ENGLISH   | ❌ OTHER VARIATIONS   |
| --------------------- | --------------------- |
| Center, fiber         | Centre, fibre         |
| Recognize, prioritize | Recognise, prioritise |
| Favorite, color       | Favourite, colour     |

We do not accept contributions to Scaleway Documentation in languages other than English.

### Tone

The tone of Scaleway Documentation is generally **formal**. Do not use the **passive voice**, **"please"** or **"thank you"**, **the future tense** or **contractions**.

| ✅ DO WRITE                                | ❌ DON'T WRITE                                  |
| ------------------------------------------ | ----------------------------------------------- |

> | Click **DELETE**  to delete the Instance.  | The Instance is deleted by clicking **DELETE**. |

Reviewer: ```suggestion
| Click **DELETE** to delete the Instance.  | The Instance is deleted by clicking **DELETE**. |
```

| Concepts           | Defines and briefly explains the main terminology associated with a Scaleway product, e.g. [Kubernetes Concepts](/containers/kubernetes/concepts/)                                            |
| Troubleshooting    | Gives solutions to help resolve a common issue with a Scaleway product, e.g. [VPC auto-config is not working](/network/vpc/troubleshooting/autoconfig-not-working/)                          |
| API/CLI          | Presents information and tips for creating and configuring a Scaleway product via both official and third-party APIs, CLIs, SDKs and other developer tools, e.g. Terraform, e.g. [Using Object Storage with the AWS-CLI](/storage/object/api-cli/object-storage-aws-cli/). These pages complement the [main developer reference documentation](https://www.scaleway.com/en/developers/).                           |
| Additional Content | Provides detailed information on specific subjects and features associated with a Scaleway product, e.g. [Optimize your Object Storage performance](/storage/object/reference-content/optimize-object-storage-performance/)                                            |

> | Tutorial           | Explains how to use a Scaleway product with a third-party tool, or how to use multiple Scaleway products together for a specific use-case, e.g. [Setting up a Proxmox cluster on Elastic Metal servers using Private Networks](/tutorials/setup-elastic-metal-proxmox-cluster-with-private-networks/)                                            |

Reviewer: Maybe put another example here, this tuto is very complex and maybe not the best example

Reviewer: ```suggestion
| Tutorial           | Explains how to use a Scaleway product with a third-party tool, or how to use multiple Scaleway products together for a specific use-case, e.g. [Creating WordPress Instances with the Scaleway CLI
](/tutorials/create-wordpress-instances-cli/)                                            |
```

### Numbered steps

When explaining to a user what to do, use **numbered steps**.

- Each step should represent one action the user must do.
- Each step should start with an imperative verb.
- If you want to explain the effect of the action, do so in a sentence at the end of the numbered step, or on a new, non-numbered indented line.

✅ **GOOD**

1. Click **Instances** in the **Compute** section of the Scaleway console. The **Instances dashboard** displays.
2. Click the Instance whose flexible IP address you want to detach. The **Overview page** for that Instance displays.
3. Click **Detach** <Icon name="unlink" /> next to the flexible IP address you want to detach.

❌ **BAD**

1. Start by clicking **Instances** in the **Compute** section of the Scaleway console.
2. You will see the **Instances dashboard**.
3. Next, you need to find the Instance whose flexible IP address you want to detach, and click on it. You'll see the **Overview page** for the Instance. To finish the process, click **Detach** <Icon name="unlink" /> next to the flexible IP address you want to detach.

> 

Reviewer 2: Maybe we can put the tip under the "numbered steps" part (after the list), as I first thought the box was part of the "Bad" section

When linking to an external page, copy and paste the full link exactly as it appears in the browser (including any slashes as shown).

When writing a "fake" or example link that does not lead to a real page, put it in inline code by using backticks: ```` `www/example-link.com` ````.

When including a "fake" or example link in a code box with an `<a href>` tag or similar, include the keyword `URLexample` in the URL: 

````
```php
<?php
more code here
?>
 
To go to the next page <A HREF="URLexample/nextpage.php">click here</A>
```
````

### Inline console icons

Use a console icon when referring in the documentation to the equivalent icon in the Scaleway console.

| Syntax                                                              | Renders as                                                |
| ------------------------------------------------------------------- | ---------------------------------------------------------- |
| ````Use the toggle <Icon name="toggle" /> to activate the feature````  | Use the toggle <Icon name="toggle" /> to activate the feature |
 
See the [full list of available icons](/#inline-console-icons).

### Messages

> Three types of message box are available, to use as follows:

Reviewer 2: ```suggestion
Three types of message boxes are available, to use as follows:
```

### Console wording conventions

When referring to different aspects of the Scaleway console, use the following terms:

| Console element             | Screenshot                                                     |
|-----------------------------|----------------------------------------------------------------|
| Side menu                   | <Lightbox src="scaleway-console-side-menu.webp" alt=""  />     |
| Organization drop-down menu | <Lightbox src="scaleway-org-dropdown.webp" alt=""  />          |
| $PRODUCT_NAME dashboard, e.g. Object Storage dashboard | <Lightbox src="scaleway-console-dashboard.webp" alt=""  />  |
| $PRODUCT_NAME creation page ( NOT `splashscreen`), e.g. Instances creation page | <Lightbox src="scaleway-instances-creation-page.webp" alt=""  /> | 
| $PRODUCT_NAME creation wizard, e.g. Load Balancer creation wizard | | <Lightbox src="scaleway-creation-wizard.webp" alt=""  /> |           

### Product and feature names

When referring to Scaleway products and features, follow the wording and capitalization that is used on the Scaleway console. See also the [capitalization rules](#capitalization).

> A wording handbook is available internally, with a full list of correct product and feature naming. Contact a member of the Scaleway Documentation Team for more details.

Author: ```suggestion
A wording handbook is available internally, with a full list of correct product and feature naming. Contact a member of the Scaleway Documentation team for more details.
```

### Page format

All Scaleway documentation pages should be `.mdx` files (MDX is an extension to Markdown that lets you include JSX). Any images included within the page must be `.webp` files.

### Page structure

Scaleway Documentation pages should conform to the following structure:

| PAGE TYPE          | DESCRIPTION                                                                              |
| -------------------| ---------------------------------------------------------------------------------------- |
| [Metadata](#metadata)           | At the top of the page, used for indexing, search and the generation of the page title (H1). The `dates` section is crucial for tracking and displaying when pages were first created and last updated. |
| [Introduction](#introduction)       | A few sentences (or short paragraphs), that briefly explain the purpose of the documentation and give any necessary preliminary information. This will be displayed in search results for the page in Scaleway's internal search engine. |

> | [Requirements](#requirements)       | Bullet points stating what the user must already have accomplished before starting any instructions on this documentation page. Necessary for How tos, Quickstarts and Tutorials. May not be relevant for other page types, e.g. FAQ, Concepts. |

Reviewer 3: nevermind, just saw that you did at the end, maybe link to that anchor?

| Page body       | The content depends on the [page type](#page-type). Split long pages into subsections with [headers](#headers). Always use [numbered lists](#numbered-steps) when explaining steps a user should take. See also [page content](#page-content) |

If you are creating a new page, it may be easiest to duplicate an existing page and adapt it to your requirements. See the raw files on our GitHub repo for this purpose, for example [this tutorial](https://raw.githubusercontent.com/scaleway/docs-content/main/tutorials/install-openvpn/index.mdx).

More information about the different sections of each page can be found by following the links in the table above.

## Page content 

### Headers

> - There can be only one H1 on each page (the page title, defined in the [metadata](#metadata-example-and-guidance))

Author: ```suggestion
- There can be only one H1 on each page (the page title, defined in the [metadata](#metadata))
```

❌ **BAD**

1. Start by clicking **Instances** in the **Compute** section of the Scaleway console.
2. You will see the **Instances dashboard**.
3. Next, you need to find the Instance whose flexible IP address you want to detach, and click on it. You'll see the **Overview page** for the Instance. To finish the process, click **Detach** <Icon name="unlink" /> next to the flexible IP address you want to detach.

### Code blocks

Put code snippet into code blocks, using the triple backspace character. Specify the language of the code block, using a [supported language](https://prismjs.com/#supported-languages).

Markdown syntax:

````
```python
my_list=[1,6,9,4]
my_list[1]=3 
print(my_list)
```
```` 

Renders as:

```python
my_list=[1,6,9,4]
my_list[1]=3 
print(my_list)
```

### Inline code

Use inline code to reference information in your text, such as environment variables, placeholder values, and single-line short code snippets. 

Markdown syntax (backticks):

````
Do not forget to replace `<YOUR INSTANCE NAME>` with the name of your Instance.  
```` 

Renders as:

Do not forget to replace `<YOUR INSTANCE NAME>` with the name of your Instance.

### Links

> Use native markdown syntax for links:

Author: ```suggestion
Use native Markdown syntax for links:
```

| Markdown syntax                                                     | Renders as                                                 |
| ------------------------------------------------------------------- | ---------------------------------------------------------- |
| ````Open the [Scaleway console](https://console.scaleway.com/)````  | Open the [Scaleway console](https://console.scaleway.com/) |

- Include links to **other Scaleway Documentation pages** to:
  - Avoid repeating instructions that already exist on an existing page
  - Help the user find other useful and related content
  - Link to concept definitions that may not be immediately comprehensible
- Include links to **external pages** to:
  - Help the user find third-party tools and resources that are mentioned in documentation

>   - Provide information that is not already contained within Scaleway Documentation. Favour sites such as Wikipedia, which are stable, neutral and open-source.

Author: ```suggestion
  - Provide information that is not already contained within Scaleway Documentation. Favor sites such as Wikipedia, which are stable, neutral and open-source.
```

When linking to an external page, copy and paste the full link exactly as it appears in the browser (including any slashes as shown).

When writing a "fake" or example link that does not lead to a real page, put it in inline code by using backticks: ```` `www/example-link.com` ````.

When including a "fake" or example link in a code box with an `<a href>` tag or similar, include the keyword `URLexample` in the URL: 

````
```php
<?php
more code here
?>
 
To go to the next page <A HREF="URLexample/nextpage.php">click here</A>
```
````

### Inline console icons

Use a console icon when referring in the documentation to the equivalent icon in the Scaleway console.

| Syntax                                                              | Renders as                                                |
| ------------------------------------------------------------------- | ---------------------------------------------------------- |
| ````Use the toggle <Icon name="toggle" /> to activate the feature````  | Use the toggle <Icon name="toggle" /> to activate the feature |
 

> See the [full list of available icons](/#inline-console-icons).

Author: ```suggestion
See the [full list of available icons](/components/docs/#inline-console-icons).
```

| megabits per second   | Mbps              | Bandwidth: 100 Mbps                      |
| gigabute-seconds      | GB-s              | Storage: 1 GB-s                          |

## Wording and vocabulary

### Environment variables

When citing Scaleway API or CLI commands, use the [established environment variable conventions](https://github.com/scaleway/scaleway-sdk-go/tree/master/scw#environment-variables).

### Console wording conventions

When referring to different aspects of the Scaleway console, use the following terms:

| Console element             | Screenshot                                                     |
|-----------------------------|----------------------------------------------------------------|
| Side menu                   | <Lightbox src="scaleway-console-side-menu.webp" alt=""  />     |
| Organization drop-down menu | <Lightbox src="scaleway-org-dropdown.webp" alt=""  />          |
| $PRODUCT_NAME dashboard, e.g. Object Storage dashboard | <Lightbox src="scaleway-console-dashboard.webp" alt=""  />  |
| $PRODUCT_NAME creation page ( NOT `splashscreen`), e.g. Instances creation page | <Lightbox src="scaleway-instances-creation-page.webp" alt=""  /> | 

> | $PRODUCT_NAME creation wizard, e.g. Load Balancer creation wizard | | <Lightbox src="scaleway-creation-wizard.webp" alt=""  /> |           

Reviewer 3: is there supposed to be an extra one here?

Author: Well spotted! Thanks

Continue by listing the necessary products, resources or items users must have. Remember to start each item in a way that it forms a coherent sentence when combined with the intro macro (`To complete the actions presented below, you must have: ...`).

> | DO WRITE ✅ | DO NOT WRITE ❌|

Reviewer: ```suggestion
| ✅ DO WRITE | ❌ DO NOT WRITE |
```
